pkg/server/status.go

```diff
@@ -2881,9 +2881,10 @@ func (s *systemStatusServer) HotRangesV2(
 		}
 		if local {
 			resp, err := s.localHotRanges(ctx, tenantID, requestedNodeID)
-			response.Ranges = append(response.Ranges, resp.Ranges...)
 			if err != nil {
 				response.ErrorsByNodeID[requestedNodeID] = err.Error()
```

Comment:
Could we instead return after this? Seems no point to continuing, since we return in the success condition.

```
return response, err
```

Reply:
Changed to just return the err and not a response object in the case that an error is returned. The node iterator will handle mapping the errors and responses to the final response object.